src/python/pants/engine/unions.py

```diff
@@ -72,6 +72,14 @@ def from_rules(cls, rules: Iterable[UnionRule]) -> UnionMembership:
         mapping: DefaultDict[type, OrderedSet[type]] = defaultdict(OrderedSet)
         for rule in rules:
             mapping[rule.union_base].add(rule.union_member)
+        # Subclassed union bases should inherit the superclass's union members.
```

Comment:
This seems like a separate change?

Reply:
Separate from what?

Reply:
From the main change with plugin fields being inherited. I don't see why unions are involved?

Reply:
Ah. 

Otherwise union.get(SomeTarget.PluginFields) wouldn't give you anything when you register a plugin field on Target.